app/models/user.rb

```diff
@@ -24,6 +24,16 @@ def starred_project?(project)
     starred_projects.include?(project)
   end
 
+  def self.to_csv
+    @users = User.order(:id)
+    CSV.generate do |csv|
+      csv << ["id","name","email","role"]
+      @users.each do |user|
+        csv << [user.id, user.name, user.email, user.role.name]
+      end
```

Comment:
Probably better to use the batching like above just in case people go crazy creating users: http://guides.rubyonrails.org/active_record_querying.html#retrieving-multiple-objects-in-batches